webhooks/pod_webhook.go

```diff
@@ -149,75 +112,236 @@ func (m *PodMutator) Handle(ctx context.Context, req admission.Request) admissio
 	}
 
 	// merge any provided flagd specs
-	flagSourceConfigurationSpec, err := corev1alpha1.NewFlagSourceConfigurationSpec()
+	flagSourceConfigurationSpec, err := v1alpha1.NewFlagSourceConfigurationSpec()
 	if err != nil {
 		m.Log.V(1).Error(err, "unable to parse env var configuration", "webhook", "handle")
 		return admission.Errored(http.StatusBadRequest, err)
 	}
 
-	for _, fcName := range fcNames {
-		ns, name := parseAnnotation(fcName, req.Namespace)
+	for _, fscName := range fscNames {
+		ns, name := parseAnnotation(fscName, req.Namespace)
 		if err != nil {
-			m.Log.V(1).Info(fmt.Sprintf("failed to parse annotation %s error: %s", fcName, err.Error()))
+			m.Log.V(1).Info(fmt.Sprintf("failed to parse annotation %s error: %s", fscName, err.Error()))
 			return admission.Errored(http.StatusBadRequest, err)
 		}
-		fc := m.getFlagSourceConfiguration(ctx, name, ns)
-		if reflect.DeepEqual(fc, corev1alpha1.FlagSourceConfiguration{}) {
-			m.Log.V(1).Info(fmt.Sprintf("FlagSourceConfiguration could not be found for %s", fcName))
+		fc := m.getFlagSourceConfiguration(ctx, ns, name)
+		if reflect.DeepEqual(fc, v1alpha1.FlagSourceConfiguration{}) {
+			m.Log.V(1).Info(fmt.Sprintf("FlagSourceConfiguration could not be found for %s", fscName))
 			return admission.Errored(http.StatusBadRequest, err)
 		}
 		flagSourceConfigurationSpec.Merge(&fc.Spec)
 	}
 
-	ffConfigs := []*corev1alpha1.FeatureFlagConfiguration{}
-	for _, ffName := range ffNames {
-		ns, name := parseAnnotation(ffName, req.Namespace)
-		if err != nil {
-			m.Log.V(1).Info(fmt.Sprintf("failed to parse annotation %s error: %s", ffName, err.Error()))
-			return admission.Errored(http.StatusBadRequest, err)
-		}
-		// Check to see whether the FeatureFlagConfiguration has service or sync overrides
-		ff := m.getFeatureFlag(ctx, name, ns)
-		if reflect.DeepEqual(ff, corev1alpha1.FeatureFlagConfiguration{}) {
-			m.Log.V(1).Info(fmt.Sprintf("FeatureFlagConfiguration could not be found for %s", ffName))
-			return admission.Errored(http.StatusBadRequest, err)
+	// maintain backwards compatibility of the openfeature.dev/featureflagconfiguration annotation
+	ffConfigAnnotation, ffConfigAnnotationOk := pod.GetAnnotations()["openfeature.dev/featureflagconfiguration"]
```

Comment:
could benefit from `openfeature.dev/featureflagconfiguration` being a constant

Reply:
Introduced as individual constants (annotation prefix + suffix for each of the 3 annotations)